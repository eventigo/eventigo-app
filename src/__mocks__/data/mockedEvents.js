/* @flow */
import type { Event } from '../../types/model';

const mockedEvents: Event[] = [
  {
    id: '7f0386b06616c2db3ca022b34875ad7d',
    name: 'Devel 2017',
    description: 'Lorem ipsum',
    url: 'https://example.org',
    start: '2017-04-24T10:00+01:00',
    end: '2017-04-24T12:00+01:00',
    venue: 'Praha',
    country: 'CZ',
    image: 'https://i.imgsafe.org/a5b1b555fc.png',
    tags: [
      {
        code: 'programovani',
        name: 'Programování',
        rate: 3
      },
      {
        code: 'konference',
        name: 'Konference',
        rate: 5
      },
      {
        code: 'javascript',
        name: 'JavaScript',
        rate: 3
      },
      {
        code: 'php',
        name: 'PHP',
        rate: 2
      }
    ]
  },
  {
    id: 'e32f3aa0ea2ed4935bc747659d2104bd',
    name: 'Machine Learning Prague 2017',
    description: `Machines can learn. Incredibly fast. Faster than you. They are getting smarter and smarter every day. They are already changing your world, your business and your life. Artificial intelligence revolution is here. Come and learn how to turn this threat into your biggest opportunity. This is not another academic conference. Our goal is to foster discussion between machine learning practitioners and all people who are interested in applications of modern trends in artificial intelligence. You can look forward to inspiring people, algorithms, data, applications, workshops and a lot of fun during three days as well as at two great parties.`,
    url: 'https://example.org',
    start: '2017-04-26T00:00+01:00',
    end: '2017-04-27T00:00+01:00',
    image: 'https://i.imgsafe.org/030bbc1041.jpg',
    tags: [
      {
        code: 'programovani',
        name: 'Programování',
        rate: 3
      },
      {
        code: 'konference',
        name: 'Konference',
        rate: 4
      },
      {
        code: 'machinelearning',
        name: 'Machine Learning',
        rate: 5
      },
      {
        code: 'ai',
        name: 'Umělá inteligence',
        rate: 1
      },
      {
        code: 'networking',
        name: 'Networking',
        rate: 2
      }
    ]
  },
  {
    id: 'd9b566c6dd9fe95e3559582f9f1882a7',
    name: 'Wisephora',
    description: 'Lorem ipsum',
    url: 'https://www.facebook.com/events/258779314553712/',
    start: '2017-05-30T00:00+02:00',
    image: 'https://i.imgsafe.org/e89935b30c.png',
    tags: []
  },
  {
    id: 'd3382a28fba675b49655f1500196c1e3',
    name: 'Barcamp Brno 2017',
    description: 'Lorem ipsum',
    url: 'https://www.facebook.com/events/177998046038693/',
    start: '2017-05-30T09:00+02:00',
    image:
      'https://scontent.xx.fbcdn.net/v/t1.0-9/s720x720/17309161_1468886766478209_1697312904965273491_n.jpg?oh=639ef12fccff2e58e3fadd0ca46fdb48&oe=5970B9B1',
    tags: []
  },
  {
    id: 'e6ec698d98d588f451586135d08e77e3',
    name: 'Node.js #24 Berlin',
    description:
      '<p><b>Talk #1: Large-Scale Container-Based Architectures</b><br/>Igor Drobiazko, CTO of elastic.io, will share their own experience with using Node.js for large-scale application and data integration use cases. elastic.io is a startup that has built a microservices-based integration platform as a service with Node.js and currently starts[masked] containers every month.</p>',
    url:
      'http://eventigo.cz/redirect/?url=https%3A%2F%2Fwww.meetup.com%2FNode-js-Meetup-Berlin%2Fevents%2F239795412%2F',
    start: '2017-07-18T18:45:00+02:00',
    end: null,
    venue: 'Berlin',
    country: 'DE',
    image: 'https://i.imgur.com/FxRuW15.png',
    tags: [
      {
        code: 'programovani',
        name: 'Programování - vše',
        rate: 2
      },
      {
        code: 'nodejs',
        name: 'Node.js',
        rate: 3
      }
    ]
  }
];

export { mockedEvents };
