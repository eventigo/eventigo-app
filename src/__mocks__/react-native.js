const rn = require('react-native');
jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(() => Promise.resolve())
  };
});

jest.mock('TouchableHighlight', () => 'TouchableHighlight');
jest.mock('TouchableOpacity', () => 'TouchableOpacity');

module.exports = rn;
