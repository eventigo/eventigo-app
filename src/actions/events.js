/* @flow */

import type { Action } from '../types/actions';
import type { EventsData } from '../types/apiData';

export function fetchData(): Action {
  return {
    type: 'EVENTS_FETCHING_DATA'
  };
}

export function fetchDataSuccess(data: EventsData): Action {
  return {
    type: 'EVENTS_FETCHING_DATA_SUCCESS',
    data
  };
}

export function fetchDataFailure(error: any): Action {
  return {
    type: 'EVENTS_FETCHING_DATA_FAILURE',
    errorMessage: error
  };
}
