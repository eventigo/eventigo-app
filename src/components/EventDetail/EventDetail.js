/* @flow */

import React from 'react';
import { View, Text } from 'react-native';
import { navigationHeader } from '../../styles';

import type { Event } from '../../types';

type Props = {
  navigation: {
    state: {
      params: {
        event: Event
      }
    }
  }
};

class EventDetail extends React.Component {
  props: Props;

  static navigationOptions = {
    title: ({ state }) => state.params.event.name,
    header: {
      tintColor: navigationHeader.headerTintColor,
      style: {
        backgroundColor: navigationHeader.headerBackgroundColor
      }
    }
  };

  render() {
    return (
      <View>
        <Text>{this.props.navigation.state.params.event.name}</Text>
      </View>
    );
  }
}

export default EventDetail;
