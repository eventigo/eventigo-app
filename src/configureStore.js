/* @flow */
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistStore, autoRehydrate } from 'redux-persist';
import { AsyncStorage } from 'react-native';
import reducers from './reducers';
import { createEpicMiddleware, combineEpics } from 'redux-observable';
import {
  fetchEventsEpic,
  foregroundStateEpic,
  requestReviewEpic
} from './epics';
import applyAppStateListener from 'redux-enhancer-react-native-appstate';

const rootEpic = combineEpics(
  fetchEventsEpic,
  foregroundStateEpic,
  requestReviewEpic
);
const middlewares = [createEpicMiddleware(rootEpic)];
const enhancers = composeWithDevTools(
  {
    // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md
  }
)(applyAppStateListener(), applyMiddleware(...middlewares), autoRehydrate());

export default function configureStore(initialState: any = {}) {
  const store = createStore(reducers, initialState, enhancers);

  // Get stored state, fire a rehydrate action, and begin persisting redux state
  persistStore(store, { storage: AsyncStorage }, () => {
    console.log('Redux store restored');
  });

  // Allow reducers hot reloading
  if (module.hot) {
    //$FlowFixMe
    module.hot.accept(() => {
      store.replaceReducer(require('./reducers').default);
    });
  }

  return store;
}
