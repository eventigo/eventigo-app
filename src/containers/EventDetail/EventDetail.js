/* @flow */

import React from 'react';
import { View, ScrollView, Text, Image, StyleSheet } from 'react-native';
import { Button, Grid, Col } from 'react-native-elements';
import EventTags from '../../components/EventTags';
import EventDate from '../../components/EventDate';
import Bookmark from '../../components/Bookmark';
import { navigationHeader, fontSizes, colors } from '../../styles';
import { tracker } from '../../ga';
import { addEventToCalendar, openUrl } from '../../utils';
import HTMLView from 'react-native-htmlview';

import type { Event } from '../../types/model';
import type { BookmarksState } from '../../reducers/bookmarks';
import type { NavigationStackScreenOptions } from 'react-navigation/src/TypeDefinition.js';

type Props = {
  navigation: {
    state: {
      params: {
        event: Event
      }
    }
  },
  bookmarks: BookmarksState,
  onBookmarkPress: string => any,
  onOpenSource: void => void,
  onAddToCalendar: void => void
};

class EventDetail extends React.Component {
  props: Props;

  static navigationOptions = ({ navigation }) =>
    ({
      title: navigation.state.params.event.name,
      headerTintColor: navigationHeader.headerTintColor,
      headerStyle: {
        backgroundColor: navigationHeader.headerBackgroundColor
      }
    }: NavigationStackScreenOptions);

  handleOpen = () => {
    const { url } = this.props.navigation.state.params.event;
    openUrl(url);
    tracker.trackEvent('Detail', 'Open event source');
    this.props.onOpenSource();
  };

  handleAddToCalendar = () => {
    addEventToCalendar(this.props.navigation.state.params.event);
    tracker.trackEvent('Detail', 'Add to calendar');
    this.props.onAddToCalendar();
  };

  handleBookmarkPress = (id: string) => {
    this.props.onBookmarkPress(id);
  };

  render() {
    const { event } = this.props.navigation.state.params;
    const { bookmarks } = this.props;

    return (
      <ScrollView>
        <Image source={{ uri: event.image }} style={styles.image}>
          <Bookmark
            active={bookmarks[event.id]}
            containerStyle={styles.bookmark}
            size={30}
            onPress={() => this.handleBookmarkPress(event.id)}
          />
        </Image>
        <View style={styles.body}>
          <Text style={styles.title}>
            {event.name}
          </Text>
          <EventDate event={event} style={styles.date} />
          <EventTags tags={event.tags} style={styles.tags} />
          <Grid containerStyle={styles.buttons}>
            <Col>
              <Button
                onPress={this.handleOpen}
                icon={{ name: 'external-link', type: 'font-awesome' }}
                title="OTEVŘÍT"
                buttonStyle={styles.button}
                backgroundColor={colors.secondary}
              />
            </Col>
            <Col>
              <Button
                onPress={this.handleAddToCalendar}
                icon={{ name: 'calendar-plus-o', type: 'font-awesome' }}
                title="ULOŽIT"
                buttonStyle={styles.button}
                backgroundColor={colors.secondary}
              />
            </Col>
          </Grid>

          <HTMLView
            value={event.description}
            style={styles.description}
            stylesheet={htmlStyles}
            paragraphBreak=""
            textComponentProps={{
              style: {
                color: colors.dark
              }
            }}
          />

        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  image: {
    height: 180,
    flexDirection: 'column',
    justifyContent: 'flex-end'
  },
  body: {
    padding: 10
  },
  title: {
    color: colors.dark,
    fontSize: fontSizes.h2,
    fontWeight: '300',
    marginBottom: 5
  },
  date: {
    color: colors.dark,
    fontSize: fontSizes.small,
    fontWeight: '300'
  },
  tags: {
    color: colors.dark,
    marginVertical: 5
  },
  description: {
    marginVertical: 5
  },
  buttons: {
    marginHorizontal: -5,
    marginVertical: 5
  },
  button: {
    marginLeft: 5,
    marginRight: 5,
    borderRadius: 3
  },
  bookmark: {
    position: 'absolute',
    right: 0,
    top: -4
  }
});

const htmlStyles = StyleSheet.create({
  a: {
    fontWeight: '300',
    color: colors.primary
  }
});

export default EventDetail;
