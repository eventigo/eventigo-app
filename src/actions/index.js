/* @flow */
import { toggleBookmark } from './bookmark';
import { fetchData, fetchDataSuccess, fetchDataFailure } from './events';
import { appLaunch, addPositiveEvent, askedForReview } from './statistics';
export {
  toggleBookmark,
  fetchData,
  fetchDataSuccess,
  fetchDataFailure,
  appLaunch,
  addPositiveEvent,
  askedForReview
};
