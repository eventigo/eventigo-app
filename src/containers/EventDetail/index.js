/* @flow */
import EventDetail from './EventDetail';
import { connect } from 'react-redux';
import { toggleBookmark } from '../../actions/bookmark';

const mapStateToProps: mapStateToProps = state => {
  return {
    bookmarks: state.bookmarks
  };
};

const mapDispatchToProps: mapDispatchToProps = dispatch => {
  return {
    onBookmarkPress: id => {
      dispatch(toggleBookmark(id));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EventDetail);
