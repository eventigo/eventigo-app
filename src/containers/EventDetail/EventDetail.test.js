/* @flow */

import 'react-native';
import React from 'react';
import EventDetail from '../EventDetail';
import configureStore from '../../configureStore';
import eventsReducer from '../../reducers/events';
import { fetchDataSuccess } from '../../actions/events';
import { mockedEvents } from '../../__mocks__/data/mockedEvents';
import { Provider } from 'react-redux';
import Config from 'react-native-config';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import type { RootState } from '../../reducers';

let store;

beforeAll(() => {
  Config.HOME_COUNTRY = 'CZ';
  const initialState: RootState = {
    events: eventsReducer(undefined, fetchDataSuccess(mockedEvents)),
    bookmarks: {},
    statistics: {
      launches: 0,
      requestedReviewDate: null,
      positiveEvents: 0
    }
  };
  store = configureStore(initialState);
});

it('renders correctly with home country event', () => {
  const navProp = {
    state: {
      params: {
        event: mockedEvents[0]
      }
    }
  };
  const tree = renderTree(store, navProp);
  expect(tree).toMatchSnapshot();
});

it('renders correctly with foreign country event', () => {
  const navProp = {
    state: {
      params: {
        event: mockedEvents[4]
      }
    }
  };
  const tree = renderTree(store, navProp);
  expect(tree).toMatchSnapshot();
});

const renderTree = (storeWithEvent, navProp) => {
  return renderer.create(
    <Provider store={storeWithEvent}>
      <EventDetail navigation={navProp} />
    </Provider>
  );
};
