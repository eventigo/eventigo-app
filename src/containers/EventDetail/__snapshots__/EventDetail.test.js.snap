// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with foreign country event 1`] = `
<RCTScrollView>
  <View>
    <Image
      source={
        Object {
          "uri": "https://i.imgur.com/FxRuW15.png",
        }
      }
      style={
        Object {
          "flexDirection": "column",
          "height": 180,
          "justifyContent": "flex-end",
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "zIndex": 99,
            },
            Object {
              "position": "absolute",
              "right": 0,
              "top": -4,
            },
          ]
        }
      >
        <View
          onLongPress={undefined}
          onPress={undefined}
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="transparent"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#2E4147",
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
    </Image>
    <View
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2E4147",
            "fontSize": 22.784,
            "fontWeight": "300",
            "marginBottom": 5,
          }
        }
      >
        Node.js #24 Berlin
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2E4147",
            "fontSize": 14.224,
            "fontWeight": "300",
          }
        }
      >
        Tuesday 18. 7. 2017 18:45
        , Berlin
        , DE
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#ACACAC",
              "fontSize": 12.64,
            },
            Object {
              "color": "#2E4147",
              "marginVertical": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          #
          programovani
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          #
          nodejs
           
        </Text>
      </Text>
      <View
        activeOpacity={1}
        containerStyle={
          Object {
            "marginHorizontal": -5,
            "marginVertical": 5,
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
            },
            Object {
              "marginHorizontal": -5,
              "marginVertical": 5,
            },
          ]
        }
      >
        <View
          activeOpacity={1}
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
              },
              undefined,
            ]
          }
        >
          <TouchableHighlight
            Ripple={undefined}
            SelectableBackground={undefined}
            SelectableBackgroundBorderless={undefined}
            activeOpacity={undefined}
            background={undefined}
            delayLongPress={undefined}
            delayPressIn={undefined}
            delayPressOut={undefined}
            disabled={false}
            hitSlop={undefined}
            onHideUnderlay={undefined}
            onLayout={undefined}
            onLongPress={undefined}
            onPress={[Function]}
            onPressIn={undefined}
            onPressOut={undefined}
            onShowUnderlay={undefined}
            underlayColor="transparent"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "marginRight": 15,
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#319F97",
                  },
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  Object {
                    "borderRadius": 3,
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "marginRight": 10,
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": undefined,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                OTEVŘÍT
              </Text>
            </View>
          </TouchableHighlight>
        </View>
        <View
          activeOpacity={1}
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
              },
              undefined,
            ]
          }
        >
          <TouchableHighlight
            Ripple={undefined}
            SelectableBackground={undefined}
            SelectableBackgroundBorderless={undefined}
            activeOpacity={undefined}
            background={undefined}
            delayLongPress={undefined}
            delayPressIn={undefined}
            delayPressOut={undefined}
            disabled={false}
            hitSlop={undefined}
            onHideUnderlay={undefined}
            onLayout={undefined}
            onLongPress={undefined}
            onPress={[Function]}
            onPressIn={undefined}
            onPressOut={undefined}
            onShowUnderlay={undefined}
            underlayColor="transparent"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "marginRight": 15,
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#319F97",
                  },
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  Object {
                    "borderRadius": 3,
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "marginRight": 10,
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": undefined,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                ULOŽIT
              </Text>
            </View>
          </TouchableHighlight>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2E4147",
            "marginVertical": 5,
          }
        }
      >
        &lt;p&gt;&lt;b&gt;Talk #1: Large-Scale Container-Based Architectures&lt;/b&gt;&lt;br/&gt;Igor Drobiazko, CTO of elastic.io, will share their own experience with using Node.js for large-scale application and data integration use cases. elastic.io is a startup that has built a microservices-based integration platform as a service with Node.js and currently starts[masked] containers every month.&lt;/p&gt;
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders correctly with home country event 1`] = `
<RCTScrollView>
  <View>
    <Image
      source={
        Object {
          "uri": "https://i.imgsafe.org/a5b1b555fc.png",
        }
      }
      style={
        Object {
          "flexDirection": "column",
          "height": 180,
          "justifyContent": "flex-end",
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "zIndex": 99,
            },
            Object {
              "position": "absolute",
              "right": 0,
              "top": -4,
            },
          ]
        }
      >
        <View
          onLongPress={undefined}
          onPress={undefined}
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="transparent"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#2E4147",
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
    </Image>
    <View
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2E4147",
            "fontSize": 22.784,
            "fontWeight": "300",
            "marginBottom": 5,
          }
        }
      >
        Devel 2017
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2E4147",
            "fontSize": 14.224,
            "fontWeight": "300",
          }
        }
      >
        Monday 24. 4. 2017 11:00 - 13:00
        , Praha
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#ACACAC",
              "fontSize": 12.64,
            },
            Object {
              "color": "#2E4147",
              "marginVertical": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          #
          programovani
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          #
          konference
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          #
          javascript
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          #
          php
           
        </Text>
      </Text>
      <View
        activeOpacity={1}
        containerStyle={
          Object {
            "marginHorizontal": -5,
            "marginVertical": 5,
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
            },
            Object {
              "marginHorizontal": -5,
              "marginVertical": 5,
            },
          ]
        }
      >
        <View
          activeOpacity={1}
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
              },
              undefined,
            ]
          }
        >
          <TouchableHighlight
            Ripple={undefined}
            SelectableBackground={undefined}
            SelectableBackgroundBorderless={undefined}
            activeOpacity={undefined}
            background={undefined}
            delayLongPress={undefined}
            delayPressIn={undefined}
            delayPressOut={undefined}
            disabled={false}
            hitSlop={undefined}
            onHideUnderlay={undefined}
            onLayout={undefined}
            onLongPress={undefined}
            onPress={[Function]}
            onPressIn={undefined}
            onPressOut={undefined}
            onShowUnderlay={undefined}
            underlayColor="transparent"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "marginRight": 15,
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#319F97",
                  },
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  Object {
                    "borderRadius": 3,
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "marginRight": 10,
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": undefined,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                OTEVŘÍT
              </Text>
            </View>
          </TouchableHighlight>
        </View>
        <View
          activeOpacity={1}
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
              },
              undefined,
            ]
          }
        >
          <TouchableHighlight
            Ripple={undefined}
            SelectableBackground={undefined}
            SelectableBackgroundBorderless={undefined}
            activeOpacity={undefined}
            background={undefined}
            delayLongPress={undefined}
            delayPressIn={undefined}
            delayPressOut={undefined}
            disabled={false}
            hitSlop={undefined}
            onHideUnderlay={undefined}
            onLayout={undefined}
            onLongPress={undefined}
            onPress={[Function]}
            onPressIn={undefined}
            onPressOut={undefined}
            onShowUnderlay={undefined}
            underlayColor="transparent"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "marginRight": 15,
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#319F97",
                  },
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  Object {
                    "borderRadius": 3,
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "marginRight": 10,
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": undefined,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                ULOŽIT
              </Text>
            </View>
          </TouchableHighlight>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2E4147",
            "marginVertical": 5,
          }
        }
      >
        Lorem ipsum
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
